### PR TITLE
[opt](profile) Avoid unnecessary copies in the profile thrift.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

need this pr 
https://github.com/apache/doris/pull/34693

before
```
Create report exec status params cost 631.780ms
Create report exec status params cost 629.973ms

Getting realtime exec status cost 2s253ms
Getting realtime exec status cost 2s365ms
Getting realtime exec status cost 2s741ms
```

now

```
Create report exec status params cost 149.688ms
Create report exec status params cost 185.920ms

Getting realtime exec status cost 1s594ms
Getting realtime exec status cost 1s703ms
Getting realtime exec status cost 1s477ms
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/runtime/runtime_query_statistics_mgr.cpp [hunk 1]
```diff
@@ -126,15 +134,16 @@ TReportExecStatusParams RuntimeQueryStatiticsMgr::create_report_exec_status_para
             }
 
             TDetailedReportParams tmp;
-            tmp.__set_profile(*pipeline_profile);
+            tmp.__isset.profile = true;
+            tmp.profile = std::move(*pipeline_profile);
```
Comment (zhiqiang-hhhh) [inline be/src/runtime/runtime_query_statistics_mgr.cpp:138]: does this move safe? the original object will be broken

File: be/src/runtime/runtime_query_statistics_mgr.cpp [hunk 2]
```diff
@@ -151,15 +160,17 @@ TReportExecStatusParams RuntimeQueryStatiticsMgr::create_report_exec_status_para
             continue;
         }
 
-        load_channel_profiles_req.push_back(*load_channel_profile);
+        load_channel_profiles_req.push_back(std::move(*load_channel_profile));
```
Comment (zhiqiang-hhhh) [inline be/src/runtime/runtime_query_statistics_mgr.cpp:163]: original object will be broken

File: be/src/runtime/runtime_query_statistics_mgr.cpp [hunk 3]
```diff
@@ -106,6 +106,14 @@ TReportExecStatusParams RuntimeQueryStatiticsMgr::create_report_exec_status_para
                 fragment_id_to_profile,
         const std::vector<std::shared_ptr<TRuntimeProfileTree>>& load_channel_profiles,
         bool is_done) {
+    RuntimeProfile::Counter timer {TUnit::TIME_NS};
+    Defer _print_log([&]() {
+        LOG_INFO("Create report exec status params cost {}",
```
Comment (zhiqiang-hhhh) [inline be/src/runtime/runtime_query_statistics_mgr.cpp:111]: `LOG_WARNING` if we meet some threshold

File: be/src/service/backend_service.cpp [hunk 4]
```diff
@@ -1171,6 +1173,15 @@ void BaseBackendService::get_realtime_exec_status(TGetRealtimeExecStatusResponse
     }
 
     LOG_INFO("Getting realtime exec status of query {}", print_id(request.id));
+    RuntimeProfile::Counter get_realtime_timer {TUnit::TIME_NS};
+
+    Defer _print_log([&]() {
+        LOG_INFO("Getting realtime exec status cost {}",
```
Comment (zhiqiang-hhhh) [inline be/src/service/backend_service.cpp:1179]: `LOG_WARNING` if we meet some threshold

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run cloud_p1

File: be/src/runtime/fragment_mgr.cpp [hunk 5]
```diff
@@ -272,7 +272,7 @@ void FragmentMgr::coordinator_callback(const ReportStatusRequest& req) {
                     detailed_param.__isset.profile = true;
                     detailed_param.__isset.loadChannelProfile = false;
                     pipeline_profile->to_thrift(&detailed_param.profile);
-                    params.detailed_report.push_back(detailed_param);
+                    params.detailed_report.push_back(std::move(detailed_param));
```
Comment (HappenLee) [inline be/src/runtime/fragment_mgr.cpp:275]: check have emplace_back?

File: be/src/runtime/runtime_query_statistics_mgr.cpp [hunk 6]
```diff
@@ -151,15 +160,17 @@ TReportExecStatusParams RuntimeQueryStatiticsMgr::create_report_exec_status_para
...
 
-    if (load_channel_profiles_req.size() > 0) {
-        profile.__set_load_channel_profiles(load_channel_profiles_req);
+    if (!load_channel_profiles_req.empty()) {
+        profile.__isset.load_channel_profiles = true;
+        profile.load_channel_profiles = std::move(load_channel_profiles_req);
     }
 
     TReportExecStatusParams req;
-    req.__set_query_profile(profile);
```
Comment (HappenLee) [inline be/src/runtime/runtime_query_statistics_mgr.cpp:162]: check the thrift api have a better way？
like```req.__set_query_profile(std::move(profile)```

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/runtime/runtime_query_statistics_mgr.cpp [hunk 7]
```diff
@@ -102,10 +103,18 @@ static Status _do_report_exec_stats_rpc(const TNetworkAddress& coor_addr,
...
-        const std::unordered_map<int32, std::vector<std::shared_ptr<TRuntimeProfileTree>>>&
+        std::unordered_map<int32, std::vector<std::shared_ptr<TRuntimeProfileTree>>>
                 fragment_id_to_profile,
-        const std::vector<std::shared_ptr<TRuntimeProfileTree>>& load_channel_profiles,
-        bool is_done) {
+        std::vector<std::shared_ptr<TRuntimeProfileTree>> load_channel_profiles, bool is_done) {
+    // This function will clear the data of fragment_id_to_profile and load_channel_profiles.
+    RuntimeProfile::Counter timer {TUnit::TIME_NS};
+    Defer _print_log([&]() {
+        LOG_INFO("Create report exec status params cost {}",
```
Comment (HappenLee) [inline be/src/runtime/runtime_query_statistics_mgr.cpp:112]: why we need log each profile report time？

File: be/src/util/thrift_client.h [hunk 8]
```diff
@@ -38,6 +38,11 @@ class TTransport;
 } // namespace apache
 
 namespace doris {
+
+#define SET_THRIFT_VALUES(thrift, member, value) \
```
Comment (HappenLee) [inline be/src/util/thrift_client.h:42]: why not call MOVE_VALUE

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**